utils/nr-graphql-helpers.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+const fetch = require('node-fetch');
+const instantObservabilityCategories = require('./instant-observability-categories.json');
+
+const NR_API_URL = process.env.NR_API_URL;
+const NR_API_TOKEN = process.env.NR_API_TOKEN;
+
+/**
+ * Build body param for NR GraphQL request
+ * @param {{queryString, variables}} queryBody - query string and corresponding variables for request
+ * @returns {String} returns the body for the request as string
+ */
+const buildRequestBody = ({ queryString, variables }) =>
+  JSON.stringify({
+    ...(queryString && { query: queryString }),
+    ...(variables && { variables }),
+  });
+
+/**
+ * Send NR GraphQL request
+ * @param {{queryString, variables}} queryBody - query string and corresponding variables for request
+ * @returns {Object} An object with the results or errors of a GraphQL request
+ */
+const fetchNRGraphqlResults = async (queryBody) => {
+  let results;
+  let graphqlErrors = [];
+
+  try {
+    const body = buildRequestBody(queryBody);
+
+    const res = await fetch(NR_API_URL, {
+      method: 'POST',
+      body,
+      headers: {
+        'Content-Type': 'application/json',
+        'Api-Key': NR_API_TOKEN,
+      },
+    });
+
+    if (!res.ok) {
+      graphqlErrors.push(
+        new Error(`Received status code ${res.status} from the API`)
+      );
+    } else {
+      const { data, errors } = await res.json();
+      results = data;
+      if (errors) {
+        graphqlErrors = [...graphqlErrors, ...errors];
+      }
+    }
+  } catch (error) {
+    graphqlErrors.push(error);
+  }
+
+  return { data: results, errors: graphqlErrors };
+};
+
+/**
+ * Handle errors from GraphQL request
+ * @param {Object[]} errors  - An array of any errors found
+ * @param {String} filePath  - The path related to the validation error
+ * @returns {void}
+ */
+const translateMutationErrors = (errors, filePath) => {
+  console.error(
+    `ERROR: The following errors occurred while validating: ${filePath}`
+  );
+  errors.forEach((error) => {
+    if (error.extensions && error.extensions.argumentPath) {
+      const errorPrefix = error.extensions.argumentPath.join('/');
+
+      console.error(`- ${errorPrefix}: ${error.message}`);
+    } else {
+      console.error(`- ${error.message}`);
+    }
+  });
+};
+
+/**
+ * Builds array of corresponding categories from keywords specified in a quickstart config.yml
+ * @param {String[] | undefined} configKeywords  - An array of keywords specified in a quickstart config.yml
+ * @returns {String[] | undefined } An array of quickstart categories
+ */
+const getCategoriesFromKeywords = (configKeywords = []) => {
+  const categoriesFromKeywords = instantObservabilityCategories.reduce(
+    (acc, { associatedKeywords, value }) => {
+      if (
+        associatedKeywords.some((keyword) => configKeywords.includes(keyword))
+      ) {
+        acc.push(value);
```

Comment:
```suggestion
        acc.push(associatedKeywords);
```

Reply:
We should actually only push on the subset of `keywords` on the quickstart that also appear as part of the category. We don't necessarily want to push on the whole set of associated keywords from the category.

For example, the `Infrastructure & OS` category has both `os` and `infrastructure` `categoryTerms`. A quickstart might only include the `infrastructure` keyword, but that doesn't also mean its `os`. In this case, we'd want to detect that `infrastructure` is one of the keywords associated with a category, and pass only that keyword via `categoryTerms`.

Does that make sense?

Reply:
I'm still not convinced categoryTerms provides any value, but those changes make sense!

Reply:
Thats the only away to associate the quickstart with a category 😄 

The idea is that `keywords` is free-form text to aide is searchability. These can be anything the user wants them to be. `categoryTerms` are meant to relate to categories, which is a predefined set of values we are using as a filtering mechanism. They are similar, but vastly different scopes.

Since categories and keywords are 2 distinct things, we wanted that reflected in the set of fields. By conflating keywords and categories into a single field called `keywords`, its difficult to know which are the "free form" values and which are supposed to represent a relationship to a category. 

Does that help?

Reply:
Also to add on... why `categoryTerms` and not the categories themselves?

We wanted to have the power to compose smaller units (category terms) into a broader bucket (category). Let's say tomorrow we wanted to break apart `Infrastructure & OS` and create separate `Infrastructure` and `OS` categories. Or lets say we wanted to combine `Kubernetes & containers` with `OS`. What would that mean for existing quickstarts that are attached to these categories?

By utilizing category terms as the composable unit, this allows us to perform the above changes without needing to go update a bunch of quickstarts. Quickstarts attached to the `os` category term, would automatically get bucketed with the updated categories. No need perform a bunch of updates anytime we need to make changes to the categories.

Hopefully this helps as well understand why we chose that direction!

Reply:
Pretty sure this was my code from the [earlier fork PR](https://github.com/newrelic/newrelic-quickstarts/pull/665#discussion_r776896725). 

My understanding is that DevEn are working through these changes. Just being explicit that I'm not planning to pick up the suggested change here and trust the DevEn rock stars are working through revisions. Thanks for refining this work! 🎸 

_(but also, do let us know if you'd like DevEx to hop on here and make the changes, happy to help!)_

Reply:
changes made. @jerelmiller @aswanson-nr @tyreer if yall feel like 👀

Reply:
Changes look good to me!